### PR TITLE
Update Roulette

### PR DESCRIPTION
**Pull request by Laquett**
Give a 2/3rds chance of moving the winner up, and a 1/3rds chance of running the old algorithm.

### CODE REVIEW

File: lib/rouletteManager.js [hunk 1]
```diff
@@ -43,11 +43,27 @@ RouletteManager.prototype.start = function (duration, price, onEnd) {
...
 
-            for (var tries = 0; tries < 10 && oldSpot === newSpot; tries++) {
-                newSpot = Math.dice(queueLength * .8);
-                if (newSpot + queueLength * .2 > oldSpot) {
-                    newSpot += Math.floor(queueLength * .2);
-                }
+            var Switch = Math.dice(3);
+
+            // If they join while position one, then they can be subject to the old algorithm.
+            if (oldSpot = '1')
```
Comment (Larry1123) [inline lib/rouletteManager.js:49]: This is setting oldSpot to the string of `1`

File: lib/rouletteManager.js [hunk 2]
```diff
@@ -43,11 +43,27 @@ RouletteManager.prototype.start = function (duration, price, onEnd) {
...
+                Switch = 3;
+
+            // Give a 2/3rds chance to move up, and 1/3rds chance for the old 'high risk' roll
+            switch (Switch) {
```
Comment (Larry1123) [inline lib/rouletteManager.js:53]: Why are you trying to switch over a string when `var Switch = Math.dice(3);` returns a number?
None of the switch cases will work.

File: lib/rouletteManager.js [hunk 3]
```diff
@@ -96,4 +112,4 @@ RouletteManager.prototype.forceStop = function () {
     return true;
 };
 
-module.exports = RouletteManager;
```
Comment (Larry1123) [inline lib/rouletteManager.js:99]: How was there not a new line here to start with?

File: lib/rouletteManager.js [hunk 4]
```diff
@@ -43,11 +43,27 @@ RouletteManager.prototype.start = function (duration, price, onEnd) {
...
+
+            // If they join while position one, then they can be subject to the old algorithm.
+            if (oldSpot = '1')
+                Switch = 3;
+
+            // Give a 2/3rds chance to move up, and 1/3rds chance for the old 'high risk' roll
+            switch (Switch) { 
+                case '1':
+                case '2':
+                    for (var tries = 0; tries < 10 && oldSpot === newSpot; tries+++)
```
Comment (Larry1123) [inline lib/rouletteManager.js:56]: `tries+++`???

**Review by Larry1123 — CHANGES_REQUESTED** [review]
There are better ways to do what you were trying to do other than using a switch. A switch will only be harder to maintain later.
There are also many other issues with the changes at this time.

File: lib/rouletteManager.js (hunk 1 above)
Comment (Larry1123) [reply lib/rouletteManager.js:49]: Also just noting that if ('someString') will return true every time.

File: lib/rouletteManager.js (hunk 3 above)
Comment (netux) [reply lib/rouletteManager.js:99]: That was probably my fault, I never add a LF at the end of my files.

File: lib/rouletteManager.js (hunk 1 above)
Comment (Laquett) [reply lib/rouletteManager.js:49]: Ah yeah, intended to do if (oldSpot === 1)

File: lib/rouletteManager.js (hunk 4 above)
Comment (Laquett) [reply lib/rouletteManager.js:56]: Think I got carried away with the + okay? lol. Just make it two.

File: lib/rouletteManager.js (hunk 2 above)
Comment (Laquett) [reply lib/rouletteManager.js:53]: Then its fine to take out the Apostrophes. Thanks for double checking it.

File: lib/rouletteManager.js (hunk 3 above)
Comment (Laquett) [reply lib/rouletteManager.js:99]: Yeah, I tried to re do it, because I did nothing with this line at all; But, it still came up the second time...

**Comment by Laquett** [comment]
Forking to rewrite. Thanks for your input, Larry!

**Comment by Larry1123** [comment]
You could have just commited your fixes to where you made this PR from and the PR would have auto updated with your changes, so they could be tracked.

**Comment by Laquett** [comment]
Yeah, I made a branch from the master - so I didn't have a repository at the time. I've since forked it, and will submit a new PR.

**Pull request closed without merging**